### PR TITLE
[GCI] Add recipe for Native File Dialog

### PR DESCRIPTION
**Pull request by hometue**
Hi, please check my work. I used the scons method, is this correct or do I need to manually move the files instead? Thanks.

### CODE REVIEW

**Comment by humdingerb** [comment]
I have no clue about scons, but I see that the recipe doesn't build a proper package...

**Comment by hometue** [comment]
I had a look through and it seems that all scons do is to compile, no installation, which explains why the recipe doesn't build a proper package, so I will manually copy the files. So, just to clarify, do I just copy the library into the $developLibDir and then copy the header into $includeDir? (also, the header is called nfd.h, will there be any issues?) Thanks

**Comment by pulkomandy** [comment]
Yes, in case no support is provided, you need to perform the install steps manually. You may also need "prepareInstalledDevelLibs" after copying everything in place.

**Comment by hometue** [comment]
Made the changes. I suppose this is the right way to do it?
Also, side note, why is it that I have to run mkdir -p $includeDir and not expect it to be present?

File: dev-libs/nfd/nfd-1.0.recipe [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:scons
+	"
+
+BUILD()
+{
+	cd src/
+	scons debug=0
```
Comment (humdingerb) [inline dev-libs/nfd/nfd-1.0.recipe]: Only the building with scons should be in BUILD(), the copying around should be in INSTALL()

**Comment by pulkomandy** [comment]
the mkdir is needed because you are copying files to a package, not to the actual system directory. So you need to create any directories that need to be in the package.

**Comment by hometue** [comment]
Shifted the lines, also made sure it runs.

**Comment by mmuman** [comment]
git complains about empty lines at end of file when applying the patch. Just a warning though.
It seems you install the static library at the wrong location, probably bad variable, cf. http://ports.haiku-files.org/wiki/HaikuPorterForPM/BuildRecipes
I think the package should also PROVIDE devel:nfd

**Comment by hometue** [comment]
So...remove the end line?
Whoops, odd how it runs without it.
I added devel:nfd, but there is a warning saying that no matching provides "devel:nfd". What am I supposed to do?

**Comment by hometue** [comment]
Oh, whoops, I realize what is wrong, the library is called libnfd.a, but there is no devel:libnfd, only devel:nfd. In that case, should I rename it to devel:libnfd?

**Comment by hometue** [comment]
I made the changes, using devel:libnfd instead of devel:nfd.

**Comment by mmuman** [comment]
Right, although the lib: entry should match the devel: one, I'm not sure if we should even have one for static libs which are only provided as developer stuff since they are linked to statically...

**Comment by waddlesplash** [comment]
No, there should be no lib: provide if there is only a static library, just a devel:libblah provide.

**Comment by hometue** [comment]
Well, removed that.

**Comment by mmuman** [comment]
Looks good, let's try it...

**Comment by mmuman** [comment]
Ok, there's still a blank line at the end of the zlib file, but that's ok.

**Comment by mmuman** [comment]
Thanks!

**Pull request merged**